doc/man3/SSL_CTX_set_num_tickets.pod

```diff
@@ -21,14 +21,16 @@ SSL_CTX_get_num_tickets
 
 SSL_CTX_set_num_tickets() and SSL_set_num_tickets() can be called for a server
 application and set the number of session tickets that will be sent to the
-client after a full handshake. Set the desired value in the B<num_tickets>
-argument. Typically these functions should be called before the start of the
-handshake. They may be called later to control the number of tickets that will
-be issued on receipt of a post-handshake certificate from the client (after an
-initial full handshake). The default number of tickets is 2. The default number
-of tickets sent following a resumption handshake is 1. This cannot be changed
-using these functions. It can be reduced to 0 using custom session ticket
-callbacks (see L<SSL_CTX_set_session_ticket_cb(3)>).
+client after a full handshake. Set the desired value (which could be 0) in the
+B<num_tickets> argument. Typically these functions should be called before the
+start of the handshake. They may be called later to control the number of
+tickets that will be issued on receipt of a post-handshake certificate from the
+client (after an initial full handshake).
```

Comment:
No, I mean at any time before a post-handshake auth occurs - so not necessarily during the handshake. But you're right that it probably does only apply to the SSL handle not SSL_CTX.

Reply:
I misread the text a bit, but do we have any particular callback that happens when the PHA client certificate is received and verified, but before it is again too late to issue new tickets (updated with the client identity for resumption)? If so the text should specifically call out the callback in which this will typically be done. The PHA topic is I think sufficiently complex to warrant its own paragraph, explaining that the subsequently issued tickets will have the client's identity (certificate and verification status) stored away and available post-resumption, which explains why we re-issue tickets at that point, and so the server can control that number at any time before the client PHA is complete, but really there are probably specific times (does the server "solicit" PHA?  Is there a mid-PHA callback? ...).

Reply:
I don't think there is a callback. However, the server requests a post-handshake auth by calling SSL_verify_client_post_handshake() so as long as it is done before then it should be fine - it doesn't need to be in a callback. I'll add a paragraph about it.